Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java

```diff
@@ -270,32 +272,35 @@ public void sendLog(String log) {
 		}
 	}
 	
-	private String convertJSONtoString(JSONObject json) {
+	private String convertJSONtoString(JSONObject json) throws JSONException {
 		StringBuilder result = new StringBuilder();
 		
 		if (json != null) {
-	        Iterator<String> iter = json.keys();
-	        if (iter.hasNext()) {
-	        	boolean first = true;
-	        	do {
+            JSONArray names = json.names();
+	        if ( names != null ) {
```

Comment:
Can you stick with the Branch style convention we have and not insert spaces before and after the ( internals )?